modules/plugin/feature-pregeneralized/src/main/java/org/geotools/data/gen/PreGeneralizedFeatureSource.java

```diff
@@ -465,13 +466,14 @@ private Double getRequestedDistance(Query query) {
              * a starting geometry to simplify further.
              */
             result =  (Double) query.getHints().get(Hints.GEOMETRY_SIMPLIFICATION);
-            log.warning("Hint for geometry simplification in query, fallback to base feature" );
+            log.fine("Hint for geometry simplification in query, fallback to base feature" );
         }
         if (result == null) {
-            log.warning("No hint for geometry distance in query, fallback to base feature" );
-        }
-        else {
-            log.info("Hint geometry distance: "+result);
+            log.fine("No hint for geometry distance in query, fallback to base feature" );
+        } else {
+            if(log.isLoggable(Level.FINE)) {
+                log.fine("Hint geometry distance: " +result);
+            }
```

Comment:
My one comment here is why do you have a `if(log.isLoggable(Level.FINE)) {` wrapping the one `log.fine`  but not the other?

Reply:
Because one is building a string, the other is not (one causes garbage even if the log is not logged, but the other is just playing with a constant).

Reply:
Right, makes sense.